lib/experimental/new_embed.dart

```diff
@@ -439,21 +439,52 @@ class NewEmbed {
     editorIsBusy = true;
 
     final GistLoader loader = deps[GistLoader];
-    final gist = await loader.loadGist(id);
-    context.dartSource = gist.getFile('main.dart')?.content ?? '';
-    context.htmlSource = gist.getFile('index.html')?.content ?? '';
-    context.cssSource = gist.getFile('styles.css')?.content ?? '';
-    context.testMethod = gist.getFile('test.dart')?.content ?? '';
-    context.solution = gist.getFile('solution.dart')?.content ?? '';
-    context.hint = gist.getFile('hint.txt')?.content ?? '';
-    tabController.setTabVisibility(
-        'test', context.testMethod.isNotEmpty && _showTestCode);
-    showHintButton?.hidden = context.hint.isEmpty;
-    solutionTab.toggleAttr('hidden', context.solution.isEmpty);
-    editorIsBusy = false;
 
-    if (analyze) {
-      _performAnalysis();
+    try {
+      final gist = await loader.loadGist(id);
+      context.dartSource = gist.getFile('main.dart')?.content ?? '';
+      context.htmlSource = gist.getFile('index.html')?.content ?? '';
+      context.cssSource = gist.getFile('styles.css')?.content ?? '';
+      context.testMethod = gist.getFile('test.dart')?.content ?? '';
+      context.solution = gist.getFile('solution.dart')?.content ?? '';
+      context.hint = gist.getFile('hint.txt')?.content ?? '';
+      tabController.setTabVisibility(
+          'test', context.testMethod.isNotEmpty && _showTestCode);
+      showHintButton?.hidden = context.hint.isEmpty;
+      solutionTab?.toggleAttr('hidden', context.solution.isEmpty);
+      editorIsBusy = false;
+
+      if (analyze) {
+        _performAnalysis();
+      }
+    } on GistLoaderException catch (ex) {
+      // No gist was loaded, so the editors need to be cleared out.
```

Comment:
nit: consider imperative mood: "No gist was loaded, clear out the editors"

Reply:
Done.